velox/type/Conversions.h

```diff
@@ -418,6 +455,51 @@ struct Converter<TypeKind::VARCHAR, void, TRUNCATE, LEGACY_CAST> {
   static std::string cast(const bool& val) {
     return val ? "true" : "false";
   }
+
+  static void normalizeStandardNotation(std::string& str) {
+    if (!FLAGS_experimental_enable_legacy_cast &&
+        str.find(".") == std::string::npos && isdigit(str[str.length() - 1])) {
+      str += ".0";
+    }
+  }
+
+  static void normalizeScientificNotation(std::string& str) {
+    auto idxE = str.find('E');
+    if (idxE == std::string::npos) {
+      return;
+    }
+
+    auto endCoef = idxE - 1;
+    while (endCoef > 0 && str[endCoef] == '0') {
+      --endCoef;
+    }
+    if (endCoef == 0) {
+      return;
```

Comment:
In what situation would this condition be true? Why don't we continue to process the exponent part?

Reply:
@kagamiori Wei, several boundary checks and returns in this func will never happen, given the returned string by fmt::format(":.16E"). For example, 3.000E+05, endCoef moving to the left will never be 0, since str[endCoef] will hit '.' first. I'm adding them just for completeness. Shall I remove them? Let me remove them, for perf.